src/utilities/use-index-resource-state.ts

```diff
@@ -95,5 +95,13 @@ export function useIndexResourceState<T extends {[key: string]: unknown}>(
     [allResourcesSelected, resources, resourceIDResolver],
   );
 
+  useEffect(() => {
+    const resourceIds = resources.map((resource) => resource.id);
+    setSelectedResources((oldSelectedResources) =>
+      oldSelectedResources.filter((item) => resourceIds.includes(item)),
+    );
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [resources.length]);
```

Comment:
We originally had it for when resources change, but it went into a pretty bad loop. I was concerned about using `.length` as well just in case of matching lengths. I'll revisit.

Reply:
State was probably being set every-time the effect ran. We'll only want to set the state _if_ selected resources need to be set 😄